tests/subsys/power/power_mgmt/src/dummy_driver.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <sys/printk.h>
+#include <zephyr/types.h>
+#include "dummy_driver.h"
+
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+uint32_t device_power_state;
+
+static int dummy_open(struct device *dev)
+{
+	int ret;
+
+	printk("open()\n");
+
+	ret = device_pm_get_sync(dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	return ret;
+}
+
+static int dummy_close(struct device *dev)
+{
+	int ret;
+
+	printk("close()\n");
+	ret = device_pm_put_sync(dev);
+	if (ret == 1) {
+		printk("Async suspend request ququed\n");
+	}
+
+	return ret;
```

Comment:
Please consider the `else` conditions to avoid unexpected behavior. Here and throughout.